lib/kino.ex

```diff
@@ -193,4 +212,60 @@ defmodule Kino do
   def nothing() do
     :"do not show this result in output"
   end
+
+  @doc """
+  Starts a process under the Kino supervisor.
+
+  The process is automatically terminated when the current process
+  terminates or the current cell reevaluates.
+  """
+  @spec start_child(
+          Supervisor.child_spec()
+          | {module(), term()}
+          | module()
+        ) :: DynamicSupervisor.on_start_child()
+  def start_child(child_spec) do
+    # Starting a process that calls Kino.start_child/1 in its init
+    # would block forever, so we don't allow nesting
+    if Kino.DynamicSupervisor in Process.get(:"$ancestors", []) do
+      raise "could not start #{inspect(child_spec)} using Kino.start_child/1," <>
+              " because the current process has been started with Kino.start_child/1." <>
+              " Please move the nested start outside and pass the result as an argument to this process"
+    end
+
+    %{start: start} = child_spec = Supervisor.child_spec(child_spec, [])
+    parent = self()
+    child_spec = %{child_spec | start: {Kino, :__start_override__, [start, parent]}}
+    DynamicSupervisor.start_child(Kino.DynamicSupervisor, child_spec)
+  end
+
+  @doc false
+  def __start_override__({mod, fun, args}, parent) do
+    # We switch the group leader, so that the newly started
+    # process gets the same group leder as the caller
+    initial_gl = Process.group_leader()
+    gl = Process.info(parent)[:group_leader]
```

Comment:
Pass the group leader instead of reading it through info. This has two issues:

1. This requires a lock in the parent process
2. This is going to copy a lot of information from the parent process (but this particular problem could be addressed by using Process.info/2